clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -2641,13 +2640,11 @@ void handleFailure(Throwable throwable) {
             .collect(Collectors.toSet());
     }
 
-    private boolean handleFindCoordinatorError(FindCoordinatorResponse response, KafkaFutureImpl<?> future) {
-        Errors error = response.error();
-        if (error.exception() instanceof RetriableException) {
+    private boolean handleGroupRequestError(Errors error, KafkaFutureImpl<?> future) {
+        if (error == Errors.COORDINATOR_LOAD_IN_PROGRESS || error == Errors.COORDINATOR_NOT_AVAILABLE) {
             throw error.exception();
-        } else if (response.hasError()) {
-            future.completeExceptionally(error.exception());
-            return true;
+        } else if (error != Errors.NONE) {
+            return future.completeExceptionally(error.exception());
```

Comment:
Hmm.. I think the original logic made more sense. Even if `completeExceptionally` returns false, it's still an error, right? We would not want to then proceed to `future.complete` or the next operation.

Reply:
Thanks. Updated the PR.